polkadot/node/core/pvf/src/host.rs

```diff
@@ -1007,12 +996,26 @@ pub(crate) mod tests {
 	}
 
 	/// Creates a new PVF which artifact id can be uniquely identified by the given number.
-	fn artifact_id(descriminator: u32) -> ArtifactId {
-		ArtifactId::from_pvf_prep_data(&PvfPrepData::from_discriminator(descriminator))
+	fn artifact_id(discriminator: u32) -> ArtifactId {
+		ArtifactId::from_pvf_prep_data(&PvfPrepData::from_discriminator(discriminator))
 	}
 
-	fn artifact_path(descriminator: u32) -> PathBuf {
-		artifact_id(descriminator).path(&PathBuf::from(std::env::temp_dir())).to_owned()
+	fn artifact_path(discriminator: u32) -> PathBuf {
+		fn to_bytes(mut n: u32) -> Vec<u8> {
+			let mut bytes = vec![0; 4];
+			let mut i = 0;
+			while n > 0 {
+				bytes[i] = (n & 0xff) as u8;
+				n >>= 8;
+				i += 1;
+			}
+			bytes
+		}
```

Comment:
nit: Can we instead `.encode()` the discriminator, or something along those lines? I don't feel like reviewing this :P :D

Reply:
Yeah, this is entirely unnecessary. And it doesn't reflect how things really work.

Reply:
I'll run the hash function on the pvf code, which is also not necessary for the test to pass, but at least it reflects how things work. People sometimes would try to learn how the code works by reading the tests. It's good for that regard.